test/unit/test_visual_recognition_v4.py

```diff
@@ -0,0 +1,851 @@
+                            "collection_id",
+                        "objects": [{
+                            "score": 7.0614014,
+                            "location": {
+                                "top": 1,
+                                "left": 5,
+                                "width": 5,
+                                "height": 2
+                            },
+                            "object": "object"
+                        }, {
+                            "score": 7.0614014,
+                            "location": {
+                                "top": 1,
+                                "left": 5,
+                                "width": 5,
+                                "height": 2
+                            },
+                            "object": "object"
+                        }]
+                    }]
+                },
+                "source": {
+                    "archive_filename": "archive_filename",
+                    "filename": "filename",
+                    "type": "file",
+                    "resolved_url": "resolved_url",
+                    "source_url": "source_url"
+                },
+                "errors": {
+                    "code":
+                        "invalid_field",
+                    "message":
+                        "The date provided for `version` is not valid. Specify dates in `YYYY-MM-DD` format.",
+                    "more_info":
+                        "https://cloud.ibm.com/apidocs/visual-recognition-v4#versioning",
+                    "target": {
+                        "type": "parameter",
+                        "name": "version"
+                    }
+                },
+                "dimensions": {
+                    "width": 6,
+                    "height": 0
+                }
+            }],
+            "trace":
+                "trace",
+            "warnings": [{
+                "code": "invalid_field",
+                "more_info": "more_info",
+                "message": "message"
+            }, {
+                "code": "invalid_field",
+                "more_info": "more_info",
+                "message": "message"
+            }]
+        }
+        responses.add(responses.POST,
+                      url,
+                      body=json.dumps(response),
+                      status=200,
+                      content_type='application/json')
+
+        authenticator = IAMAuthenticator('bogusapikey')
+        service = ibm_watson.VisualRecognitionV4('YYYY-MM-DD',
+                                                 authenticator=authenticator)
+        service.set_service_url(base_url)
+
+        with open(
+                os.path.join(os.path.dirname(__file__),
+                             '../../resources/cars.zip'), 'rb') as cars:
+            detailed_response = service.analyze(
+                collection_ids='collection_id1, collection_id2',
+                features=AnalyzeEnums.Features.OBJECTS.value,
```

Comment:
features and collection_ids should be list of strings, not csv. We need to convert these into csv in the service method